src/main/java/com/endava/endabank/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,236 @@
+import com.endava.endabank.utils.MailService;
+import com.endava.endabank.utils.user.UserValidations;
+import com.google.common.annotations.VisibleForTesting;
+import com.sendgrid.Response;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.AccessDeniedException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BiFunction;
+
+@Service
+@AllArgsConstructor
+public class UserServiceImpl implements UserService {
+    private UserDao userDao;
+    private ModelMapper modelMapper;
+    private IdentifierTypeService identifierTypeService;
+    private RoleService roleService;
+    private PasswordEncoder passwordEncoder;
+    private ForgotUserPasswordTokenService forgotUserPasswordTokenService;
+    private MailService mailService;
+    private MailProperties mailProperties;
+    @Override
+    @Transactional(readOnly = true)
+    public List<UserToApproveAccountDto> usersToApprove() {
+        List<User> users = userDao.findAll();
+        return users.stream().map(this::mapToUserToApproveDto).toList();
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public User findById(Integer id) {
+        return userDao.findById(id).
+                orElseThrow(() -> new UsernameNotFoundException(Strings.USER_NOT_FOUND));
+    }
+
+    @Override
+    @Transactional
+    public User save(UserRegisterDto userDto) {
+        User user = modelMapper.map(userDto, User.class);
+        Optional<User> userExist = userDao.findByEmail(user.getEmail());
+        if (userExist.isPresent()) {
+            throw new UniqueConstraintViolationException(Strings.CONSTRAINT_EMAIL_VIOLATED);
+        }
+        userExist = userDao.findByIdentifier(user.getIdentifier());
+        if (userExist.isPresent()) {
+            throw new UniqueConstraintViolationException(Strings.CONSTRAINT_IDENTIFIER_VIOLATED);
+        }
+        Role role = roleService.findById(Permissions.ROLE_USER);
+        IdentifierType identifierType = identifierTypeService.findById(userDto.getTypeIdentifierId());
+        user.setRole(role);
+        user.setIdentifierType(identifierType);
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
+        userDao.save(user);
+        return user;
```

Comment:
Hi,

If you return the user, you are gonna return the user of line 81 instead of the saved user.